CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+# Contributing to Snoopy
+
+This doc outlines the architecture of Snoopy and how provides a step-by-step
+guide of setting it up on your machine.
+
+## Setup
+
+Snoopy is a [Yarn workspace](https://yarnpkg.com/en/docs/workspaces) that is
+composed of several separate packages. These packages are all published to NPM
+together with [Lerna](https://github.com/lerna/lerna). First install all
+dependencies with:
+
+```
+yarn
+```
+
+To build Snoopy for development, you need to compile the codebase with
+[TypeScript](https://www.typescriptlang.org/). In the root of the project run:
+
+```
+yarn build:watch
+```
+
+This will watch and incrementally build all packages. If you only want to build
+without watching, drop the `:watch`.
+
+## Running on an app
+
+Snoopy itself is useless without an App to run it on. There are several examples
+included in the Snoopy codebase, but you can run it on any React project you
+want. As an example, we will use the `02-to-do-list` example.
+
+Navigate to the example:
+
+```
+cd examples/02-to-do-list
+```
+
+Start Snoopy by running node on the entry to the CLI package:
+
+```
+node ../../packages/cli/lib/index.js
+```
+
+You should see that the server has started on some port (default 3042). You can
+navigate to `localhost:PORT` and Snoopy should be running.
+
+## Making changes
+
+When you make a change to any package that is not the UI, make sure that the
+`build:watch` command has completed its incremental complication. Otherwise your
+changes will not be run. If there are any TypeScript issues and the command
+shows an error, your latest changes will not be run. Changes to the UI can be
+made while the server is still running and Parcel will update the UI with HMR.
+
+## Architecture
+
+Currently there are 7 packages apart of Snoopy. 5 of which are published to NPM.
+
+### api
+
+Contains names of all events that are sent between server and client (UI).
+
+### cli
```

Comment:
Just curious - why does this live in a different package?
It would be nice to get a rough idea of why some things have their own package and others don't. For instance, `vs-code` package makes total sense because it's almost a separate product on its own. Likewise, a big reusable library may be worth it's own package. But are `search` and `cli` meant to be useful on their own?

Reply:
The idea was to just separate bits of logic. Each package has its own set of dependencies that don't affect the other packages. Right now you cannot use the cli/server/search packages without each other, but in the future, we might want to. Such as searching for components in the users codebase from a github bot (we wouldn't want to start and express server or have a cli). Having separate packages for all these will make this easier. Also, yarn workspaces and lerna basically makes this the same as having server/cli/search/etc... separate folders in the same package.